### PR TITLE
Add sample for .NET Core IoT and NanoFramework with BMP280 Sensor, Azure SDK and Certificates

### PR DESCRIPTION
**Pull request by ikivanc**
We can use .NET and C# to develop IoT solutions on different boards. We can use different platforms for different board types such as .NET IoT Core and .NET nanoFramework platforms.

This sample includes implementation of BMP280 sensor with .NET Core IoT on Raspberry Pi and NanoFramework, pushes data through AzureSDK using certificates to Azure IoT Hub and save all data to a Azure Blob Storage. 

This documentation is comparing two platforms and providing guidance for platform specific implementations.

## Description
This is a connected scenario of devices, sensor, Azure with device certificates on .NET IoT Core and NanoFramework.

## Motivation and Context
Sample solution and documentation for comparing .NET nanoFramework and .NET Core IoT platforms. For this solution same sample application with an identical scenario is created on both platforms, M5Stack with .NET nanoFramework and Raspberry Pi 2 (or above) with .Net Core IoT.

The sample application includes a scenario on both platforms: Every 3 seconds application reads temperature value from BMP280 sensor and turns on LED after reading the value. Then this temperature value is pushed to IoT Hub with device name and after every successful push LED will blink and turn off. Data will be handled by IoT Hub message routing and will be pushed to blob storage to store.

## How Has This Been Tested?
.NET IoT Core:
* This has been tested with .NET Core IoT on Raspberry Pi 2
* Using Adafruit BMP280
* A certificated created manually

.NET nanoFramework:
* This has been tested with M5Stack on nanoFramework
* Using Adafruit BMP280
* A certificated created manually

## Screenshots
.NET IoT Core
image
image

NanoFramework:
image
image

## Types of changes
- New Sample (adds a new sample)

### CODE REVIEW

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (Ellerbach) [inline iot/dotnet-iot-and-nanoframework/README.md:6]: maybe add dotnet nanoFramework here as well?

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (Ellerbach) [inline iot/dotnet-iot-and-nanoframework/README.md:10]: I guess it's .NET IoT, you can remove the Core everywhere.

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (Ellerbach) [inline iot/dotnet-iot-and-nanoframework/README.md:19]: Maybe you can name the constrained devices. Some like "We mean by constrained devices, Micro Controller Unit (MCU) like ESP32, STM32, NXP or TI."

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (Ellerbach) [inline iot/dotnet-iot-and-nanoframework/README.md:20]: You can add a link on the Show Dotnet article: https://devblogs.microsoft.com/dotnet/show-dotnet-running-my-net-nanoframework-for-8-years-on-a-battery/

This gives example of production products running .NET nanoFramework and connected to Azure IoT.

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (Ellerbach) [inline iot/dotnet-iot-and-nanoframework/README.md:23]: Try to avoid abbreviation like "max", better use "maximum".

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (Ellerbach) [inline iot/dotnet-iot-and-nanoframework/README.md:59]: You may add the reason as well most code is the same: API between both platforms are aligned, SDK API are aligned too and you can find Iot.Device bindings as well with aligned API. And despite the fact that .NET nanoFramework do not support (yet) generics, away and async, and is a subset of the full framework, the most important API are present and everything is aligned to make it as easy as possible to reuse a maximum of code.

File: iot/dotnet-iot-and-nanoframework/dotnet-iot/Program.cs
Comment (Ellerbach) [inline iot/dotnet-iot-and-nanoframework/dotnet-iot/Program.cs:1]: It would be great to add a MIT style header, in this case, it's clear that the code can be reused for any purpose. Same for other source code files.

**Review by Ellerbach — COMMENTED** [review]
Thanks a lot for the submission and placing in parallel .NET IoT and .NET nanoFramework code, both connecting to Azure with certificates and a clear step by step how to setup everything. Great example!

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (ikivanc) [reply iot/dotnet-iot-and-nanoframework/README.md:6]: This is a pre-defined taxonomies product list, I checked nanoFramework but it's not in pre-defined list.

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (ikivanc) [reply iot/dotnet-iot-and-nanoframework/README.md:10]: Thanks @Ellerbach this is fixed

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (ikivanc) [reply iot/dotnet-iot-and-nanoframework/README.md:19]: This is fixed

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (ikivanc) [reply iot/dotnet-iot-and-nanoframework/README.md:20]: Thanks this is added

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (ikivanc) [reply iot/dotnet-iot-and-nanoframework/README.md:23]: This is fixed

File: iot/dotnet-iot-and-nanoframework/README.md
Comment (ikivanc) [reply iot/dotnet-iot-and-nanoframework/README.md:59]: Thanks, this is added as well

File: iot/dotnet-iot-and-nanoframework/dotnet-iot/Program.cs
Comment (ikivanc) [reply iot/dotnet-iot-and-nanoframework/dotnet-iot/Program.cs:1]: Thanks this is added as well.

**Review by CamSoper — CHANGES_REQUESTED** [review]
This is great! I'm taking Thursday and Friday off, so I won't have any more time to look at this this week. I plan on looking at it in more depth Monday.

In the meantime, could you please make sure the [markdown linting errors](https://github.com/dotnet/samples/pull/4820/checks?check_run_id=3681501418) are resolved?

@dotnet/docs Don't wait on my re-review to merge this if it looks fine otherwise. Otherwise I'll get to it next week.

**Review by CamSoper — APPROVED** [review]

**Pull request merged**
